kclvm/driver/src/kpm_update.rs

```diff
@@ -0,0 +1,96 @@
+use anyhow::{bail, Result};
+use std::{collections::HashMap, env, iter, path::PathBuf, process::Command};
+
+const MANIFEST_FILE: &str = "kcl.mod";
+
+/// Update the KCL module.
+///
+/// This function calls `kcl mod update` to update the KCL module.
+pub(crate) fn update_kcl_module(manifest_path: PathBuf) -> Result<()> {
```

Comment:
You can try to add some test cases for `update_kcl_module()`

Reply:
I tried to add basic functions for test cases for the `update_kcl_module`. While I was exploring the tests.rs file I got some confusions as follows:

1. What is this `test_vendor` file for?
2. Why do we have  expand file pattern dir?
3. I am bit confused about the path passing to kcl mod update from the metadata function that we already have?

I was following the kpm_metadata test cases implementation, also wanted to know about the use case of this `test_fill_pkg_maps_for_k_file` and `test_fill_pkg_maps_for_k_file_with_line`?

cc @zong-zhe

Reply:
- `test_vendor` is a tmp dir for test, used to test whether the environment variable `$KCL_PKG_PATH` is working properly. This environment variable is used to specify the directory in which the KCL dependencies is stored.

- `test_fill_pkg_maps_for_k_file ` is used to test the function `fill_pkg_maps_for_k_file`, which will find the nearest   `kcl.mod` file from the *.k file, and get the dependency path from the `kcl.mod`.

- `test_fill_pkg_maps_for_k_file_with_line` and `test_fill_pkg_maps_for_k_file` is the similar, the difference is that `test_fill_pkg_maps_for_k_file_with_line` needs to test the special processing of underscores and horizontal lines in the name of the dependencies.